src/client/api/events/sub_messages_metadata.c

```diff
@@ -12,10 +12,10 @@ int event_subscribe_msg_metadata(event_client_handle_t client, int *mid, char co
     printf("[%s:%d]: Message Id length is invalid\n", __func__, __LINE__);
     return -1;
   }
-  // Buffer to store topic string : messages/{messageid}/metadata
-  char topic_buff[BIN_TO_HEX_BYTES(IOTA_MESSAGE_ID_BYTES) + 21] = {0};
+  // Buffer to store topic string : messages-metadata/{messageid}
+  char topic_buff[BIN_TO_HEX_BYTES(IOTA_MESSAGE_ID_BYTES) + 20] = {0};
```

Comment:
prefix `0x` is missing
```suggestion
  // Buffer to store topic string : messages-metadata/0x{messageid}
  char topic_buff[BIN_TO_HEX_BYTES(IOTA_MESSAGE_ID_BYTES) + 21] = {0};
```

Reply:
0x was considered in topic as below, but was not explicitely mentioned in the comment. We need only len 20 for messages-metadata/0x. Added 0x in comment also. Thanks.
sprintf(topic_buff, "message-metadata/0x%s", msg_id);